nifi-frontend/src/main/frontend/apps/nifi/src/app/pages/flow-designer/ui/canvas/items/flow/flow-diff-dialog/flow-diff-dialog.html

```diff
@@ -0,0 +1,123 @@
+  ~ limitations under the License.
+  -->
+
+<h2 mat-dialog-title>
+    Flow Version Diff - {{ flowName }}
+</h2>
+<div class="flow-diff">
+    <mat-dialog-content>
+        <div class="dialog-content flex flex-col gap-y-4">
+            <div
+                class="flow-diff-message"
+                data-qa="flow-diff-message"
+                *ngIf="comparisonSummary.length > 0">
+                <div class="flow-diff-message-title">Comparing versions:</div>
+                <ul class="flow-diff-message-list">
+                    <li *ngFor="let summary of comparisonSummary">
+                        <span class="flow-diff-version">Version {{ summary.version }}</span>
+                        <span *ngIf="summary.created" class="flow-diff-created">
+                            ({{ summary.created }})
+                        </span>
+                    </li>
+                </ul>
+            </div>
+
+            <div class="flow-diff-controls flex flex-wrap gap-4">
+                <mat-form-field class="flow-diff-select" appearance="outline">
+                    <mat-label>Current Version</mat-label>
+                    <mat-select [formControl]="currentVersionControl" data-qa="current-version-select">
+                        <mat-option *ngFor="let version of versionOptions" [value]="version">
+                            {{ formatVersionOption(version) }}
+                        </mat-option>
+                    </mat-select>
+                </mat-form-field>
+
+                <mat-form-field class="flow-diff-select" appearance="outline">
+                    <mat-label>Selected Version</mat-label>
+                    <mat-select [formControl]="selectedVersionControl" data-qa="selected-version-select">
+                        <mat-option *ngFor="let version of versionOptions" [value]="version">
+                            {{ formatVersionOption(version) }}
+                        </mat-option>
+                    </mat-select>
+                </mat-form-field>
+
+                <mat-form-field class="flow-diff-filter" appearance="outline">
+                    <mat-label>Filter</mat-label>
+                    <input
+                        matInput
+                        type="text"
+                        [formControl]="filterControl"
+                        placeholder="Filter differences"
+                        data-qa="flow-diff-filter" />
+                </mat-form-field>
+            </div>
+
+            <div class="flow-diff-table flex-1">
+                <div class="flow-diff-loading" *ngIf="isLoading">
+                    <mat-progress-spinner diameter="36" mode="indeterminate"></mat-progress-spinner>
+                </div>
+
+                <div *ngIf="!isLoading">
+                    <div *ngIf="errorMessage" class="flow-diff-error" data-qa="flow-diff-error">
+                        {{ errorMessage }}
+                    </div>
```

Comment:
The listings in nifi do not show errors like this. If there is an error when requesting the diff it should be displayed in a context error banner at the top of the dialog like we do in other cases.


```suggestion
```